### DISCUSSION TITLE
Why does Assert.Raises only support EventHandlers but not Actions?

### DISCUSSION THREAD

**Discussion by TimKEviden** [xunit/xunit#2899] [Question] [answered]
Hey there!

I have the case where a `public event Action<MyEventArgs>` is invoked. 
I want to write a test to ensure that it is really invoked when I need it to be invoked.
While trying to implement it in my test I did not get  to run as I always get the error 
_Cannot implicitly conevert type 'System.EventHandler\<MyEventArgs\> to System.Action\<MyEventArgs\>'_.

I am currently using version 2.7.0

Is there a reason for only EventHandlers being supported?

Best wishes!

**Comment by bradwilson** [comment, marked as answer]
No reason other than "nobody has requested it until now". 😄

**Reply by bradwilson** [reply]
https://github.com/xunit/xunit/issues/2900

**Reply by bradwilson** [reply]
This is available now in a pre-release build, and will be part of the next official release.

- Added support for `Action<T>` to `Assert.Raises<T>`, `Assert.RaisesAny<T>`, `Assert.RaisesAnyAsync<T>`, and `Assert.RaisesAsync<T>`. These versions will return `RaisedEvent<T>`, with a `null` value for `Sender`.
- Added support for `Action` to `Assert.Raises` and `Assert.RaisesAsync`. These versions do not return values, since `Action` as an event does not accept any values, and there are no senders with `Action`-based event handlers. They will simply throw or pass.

Links to the v2 and v3 builds (as well as links to the updated code and unit tests) are available in the issue.

**Comment by TimKEviden** [comment]
Oh wow perfect, thanks for the quick anser/addition to the project! 😄
